src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.cs

```diff
@@ -464,7 +476,18 @@ protected void OnCreated(FileSystemEventArgs e)
         [SuppressMessage("Microsoft.Security", "CA2109:ReviewVisibleEventHandlers", MessageId = "0#", Justification = "Changing from protected to private would be a breaking change")]
         protected void OnDeleted(FileSystemEventArgs e)
         {
-            _onDeletedHandler?.Invoke(this, e);
+            InvokeOn(e, _onDeletedHandler);
+        }
+
+        private void InvokeOn(FileSystemEventArgs e, FileSystemEventHandler handler)
+        {
+            if (handler != null)
+            {
+                if (SynchronizingObject != null && SynchronizingObject.InvokeRequired)
```

Comment:
Ah yeah, we're not using the _onXHandlers directly. 

The SynchronizingObject thing is potentially an issue [and one that is present in desktop as well](https://referencesource.microsoft.com/#System/services/io/system/io/FileSystemWatcher.cs,819). We can't exactly catch the nullreference here because it's possible that the SynchronizingObject.BeginInvoke implementation may throw that in which case we don't want to mess with it. It also doesn't seem like we would want to copy the object or lock around it either, but I'm not too familiar with this synchronization scheme.